components/prism-core.js

```diff
@@ -280,9 +280,15 @@ var _ = _self.Prism = {
 					lookbehindLength = 0,
 					alias = pattern.alias;
 
+				if (greedy && !pattern.pattern.global) {
+					// Without the global flag, lastIndex won't work
+					pattern.pattern = RegExp(pattern.pattern.source, pattern.pattern.flags + "g");
```

Comment:
@zeitgeist87: How recent is the `flags` property? It [looks like](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/flags) it's not supported very well... I just ran into errors running `npm test` and I think they are due to this.

Maybe we could refactor like so:

```javascript
var flags = pattern.pattern.toString().match(/[imuy]*$/)[0];
pattern.pattern = RegExp(pattern.pattern.source, flags + "g");
```

What do you think?